Add a testcase for escaping script element contents

tormeh opened the pull request:
…innerHTML on a script tag escapes tag characters". I'm honestly unsure if it does this correctly, but learning by doing and all that....

<!-- Please describe your changes on the following line: -->
Added new item to the test elements: <script type="test"><&></script>
and a corresponding item to the "expected" list: ["<script type=\"test\"><&><\/script>", "<script type=\"test\"><&><\/script>"]

---
<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `__` with appropriate data: -->
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes test #14975 (github issue number if applicable).

<!-- Either: -->
- [ ] There are tests for these changes OR
- [x] These changes do not require tests because it is a test

<!-- Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process. -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/15148)
<!-- Reviewable:end -->

tormeh committed:
This commit is intended to add a test case for Github issue #14975: "innerHTML on a script tag escapes tag characters". I'm honestly unsure if it does this correctly, but learning by doing and all that....

jdm commented on tests/wpt/web-platform-tests/html/syntax/serializing-html-fragments/serializing.html:
```diff
@@ -58,6 +59,7 @@
 ["\"", "<span>\"</span>"],
 ["<style><&></style>", "<span><style><&></style></span>"],
 ["<script type=\"test\"><&><\/script>", "<span><script type=\"test\"><&><\/script></span>"],
+["<script type=\"test\"><&><\/script>", "<script type=\"test\"><&><\/script>"],
```
This test performs `element.innerHTML` on each test element, so the `<script>` and `</script>` output should not be in the expected output.

jdm commented on tests/wpt/web-platform-tests/html/syntax/serializing-html-fragments/serializing.html:
```diff
@@ -25,6 +25,7 @@
 <span>&quot;</span>
 <span><style><&></style></span>
 <span><script type="test"><&></script></span>
+<script type="test"><&></script>
```
No need for the `type="test"` here, since we only care about the child contents of the element.

Ms2ger replied:
I imagine this would cause an unhandled error event when we try to execute `<&>` as JS,

jdm reviewed (changes requested):
This looks like the test I was expecting to see!

jdm commented:
Presumably `./mach test-wpt tests/wpt/web-platform-tests/html/syntax/serializing-html-fragments/serializing.html` complains about unexpected results now, since this test should not be passing. We either need to fix the issue before we can merge the test, or add a new entry to `tests/wpt/metadata/html/syntax/serializing-html-fragments/serializing.html.ini`.

tormeh committed:
Changes made!

Yeah, I have less than 10 hours of experience with javascript; sorry about the inevitable mistakes. I think these are the changes you were looking for, though.

tormeh reviewed (commented):
I think these are the changes you are looking for. I have less than 10 hours of experience with javascript; sorry about the inevitable mistakes.

jdm commented on tests/wpt/web-platform-tests/html/syntax/serializing-html-fragments/serializing.html:
```diff
@@ -59,7 +59,7 @@
 ["<style><&></style>", "<span><style><&></style></span>"],
 ["<script type=\"test\"><&><\/script>", "<span><script type=\"test\"><&><\/script></span>"],
-["<script type=\"test\"><&><\/script>", "<script type=\"test\"><&><\/script>"],
+["<script><&><\/script>", "<&>"],
```
Aren't these two values reversed? The first value is the expected output of innerHTML, the second is the expected output of outerHTML.

tormeh replied:
Can I put them both in a single PR? I thought the test being in the servo repository and the serializer being in the html5ever repository made that hard?

jdm reviewed (changes requested):
Are you planning to include the fix for html5ever in this PR before we merge the new testcase?

tormeh committed:
reversed inner/outerhtml

reversed inner/outerhtml

tormeh commented:
A bit unsure how this page looks to you, but I'll reply here as well: I didn't think the two changes could be in the same PR, since they're in different repositories: servo vs html5ever.

jdm commented:
Once the change is part of html5ever, we will need to update Servo to use the latest version of that library. We can either merge this new testcase immediately and mark this new test as an expected failure, or we can wait to merge it until we can include the change to update html5ever. Your choice!

tormeh commented:
Hm... I think it's better to add it as an expected failure. Then I can test my patch against this test.

jdm commented:
You can run `./mach test-wpt tests/wpt/web-platform-tests/html/syntax/serializing-html-fragments/serializing.html --log-raw /tmp/servo; ./mach update-wpt /tmp/servo` to do that, then commit the changes.

jdm commented:
@tormeh Are you still planning to address my last comment?

tormeh commented:
Yeah, I'm having a problem with doing that because the version of html5ever that servo uses is not the newest version, so the instructions for "working on a crate" under https://github.com/servo/servo/blob/master/docs/HACKING_QUICKSTART.md is not really working. I get compilation errors because the newest versions of servo and html5ever are incompatible. I guess I could checkout an earlier version with git but I haven't tried it yet, because I don't know how and honestly it sounds unappealing. 

I suspect Servo might be a more pleasant experience for people who are already comfortable with semi-advanced git usage and who has worked with big codebases before. I mean, I haven't yet touched any of the actual Rust code (or, well, I've written a print-statement in the html5ever source stored in the servo cargo cache but cargo doesn't really work like that, I found out). I think I'm bouncing off of this one. Sorry.

jdm commented:
Oh, you're running into the fact that https://github.com/servo/html5ever/pull/249 just merged while https://github.com/servo/servo/pull/15283 is still outstanding. Shoot; we're not supposed to do that :<

jdm commented:
I'm going to go ahead and revert that html5ever commit. Please let me know if that causes you to change your mind, or go ahead and close this PR if you don't intend to keep working on this. Sorry it wasn't a positive experience so far!

jdm commented:
I'm going to go ahead and free this up for someone else to work on.
